test/envtest/dlbdeviceplugin_controller_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envtest
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	devicepluginv1 "github.com/intel/intel-device-plugins-for-kubernetes/pkg/apis/deviceplugin/v1"
+)
+
+var _ = Describe("DlbDevicePlugin Controller", func() {
+
+	const timeout = time.Second * 30
+	const interval = time.Second * 1
+
+	AfterEach(func() {
+		time.Sleep(time.Second * 2)
+
+	})
+
+	Context("Basic CRUD operations", func() {
+		It("should handle DlbDevicePlugin objects correctly", func() {
+			spec := devicepluginv1.DlbDevicePluginSpec{
+				Image: "testimage",
+			}
+
+			key := types.NamespacedName{
+				Name: "dlbdeviceplugin-test",
+			}
+
+			toCreate := &devicepluginv1.DlbDevicePlugin{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: key.Name,
```

Comment:
Why do we set plugin name to the namespace name?

Reply:
The CRDs are cluster scoped: see 37618d4f850f63ae0503b0b313242540a9bb269d

Reply:
not sure I follow. I was proposing to test [this line of code](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/pkg/controllers/dlb/controller.go#L78)

Reply:
the namespace where the daemonset is created is not taken from the CR namepace (they don't have one) so it's not possible to test that, at least with envtest

Reply:
got it. thanks for the explanations!